### PR TITLE
Fix test

### PR DESCRIPTION
**Pull request by joaobrunoah**
# Description

Fix a test where Vault balances were not checked properly.

## Type of change

- [x] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Optimization: [ ] gas / [ ] bytecode
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- [x] Complex code has been commented, including external interfaces
- [x] Tests have 100% code coverage
- [x] The base branch is either `main`, or there's a description of how to merge

### CODE REVIEW

File: pkg/vault/test/foundry/E2eErc4626Swaps.t.sol [hunk 1]
```diff
@@ -329,12 +317,14 @@ contract E2eErc4626SwapsTest is BaseERC4626BufferTest {
         uint256 senderDaiDelta = balancesBefore.balances.bobTokens[balancesBefore.daiIdx] -
             balancesAfter.balances.bobTokens[balancesAfter.daiIdx];
         uint256 vaultTotalDaiBefore = balancesBefore.balances.vaultTokens[balancesBefore.daiIdx] +
-            waDAI.previewRedeem(balancesBefore.balances.vaultTokens[balancesBefore.waDaiIdx]);
+            waDAI.previewMint(balancesBefore.balances.vaultTokens[balancesBefore.waDaiIdx]);
         uint256 vaultTotalDaiAfter = balancesAfter.balances.vaultTokens[balancesAfter.daiIdx] +
-            waDAI.previewRedeem(balancesAfter.balances.vaultTokens[balancesAfter.waDaiIdx]);
+            waDAI.previewMint(balancesAfter.balances.vaultTokens[balancesAfter.waDaiIdx]);
```
Comment (jubeira) [inline pkg/vault/test/foundry/E2eErc4626Swaps.t.sol:322]: This just changes the rounding to up; does it make a difference in this case? Can you explain the rationale?

File: pkg/vault/test/foundry/E2eErc4626Swaps.t.sol [hunk 2]
```diff
@@ -346,17 +336,30 @@ contract E2eErc4626SwapsTest is BaseERC4626BufferTest {
...
+
+        // `vaultTotalUsdcAfter - vaultTotalUsdcBefore = senderUsdcDelta`, solve for usdcAfter to prevent underflows.
         assertApproxEqAbs(
             vaultTotalUsdcAfter,
             senderUsdcDelta + vaultTotalUsdcBefore,
             MAX_ERROR,
             "Vault usdc/waUSDC balance is wrong"
         );
+
+        // DAI and USDC worths the same. So, we can sum the deltas for DAI and USDC in the vault, and compare with
```
Comment (jubeira) [inline pkg/vault/test/foundry/E2eErc4626Swaps.t.sol:351]:
```suggestion
        // DAI and USDC are worth the same. So, we can sum the deltas for DAI and USDC in the vault, and compare with
```
```suggestion
        // DAI and USDC worths the same. So, we can sum the deltas for DAI and USDC in the vault, and compare with
```

File: pkg/vault/test/foundry/E2eErc4626Swaps.t.sol [hunk 3]
```diff
@@ -346,17 +336,30 @@ contract E2eErc4626SwapsTest is BaseERC4626BufferTest {
...
+        // how much the user paid in fees. The user cannot have any benefit, but the vault and the user amount may
+        // differ because of rounding. Let's call the sum of vault deltas as sumDeltaVault, and the sum of sender
+        // deltas as sumDeltaSender, the test belows tests if
+        // `sumDeltaSender - 2 * MAX_ERROR < sumDeltaVault < sumDeltaSender`, it means, vault delta is smaller than
+        // sender delta (sender paid more than vault received), but the difference is small.
```
Comment (jubeira) [inline pkg/vault/test/foundry/E2eErc4626Swaps.t.sol:356]: Sounds good. 

How about also checking the wrapper rate before and after the operations? It doesn't really affect the vault, but wrapper rate increase basically explains this difference at the vault level (user pays more than what the vault receives, wrapper pockets the difference).

**Review by jubeira — APPROVED** [review]
LGTM, good cleanup! Let's wrap up these final details and merge into the other branch.

File: pkg/vault/test/foundry/E2eErc4626Swaps.t.sol (hunk 1 above)
Comment (joaobrunoah) [reply pkg/vault/test/foundry/E2eErc4626Swaps.t.sol:322]: No difference at all. I changed it to test a few things and didn't change it back.

File: pkg/vault/test/foundry/E2eErc4626Swaps.t.sol (hunk 3 above)
Comment (joaobrunoah) [reply pkg/vault/test/foundry/E2eErc4626Swaps.t.sol:371]: Done!

**Pull request merged**